scripts/tests/localsetup_tests.py

```diff
@@ -1429,3 +1429,53 @@ def test_parse(self):
         for ver, want in cases:
             got = parse_version(ver)
             self.assertEqual(want, got)
+
+    @mock.patch(cli.__name__ + ".load_images")
+    def test_elasticsearch_tls(self, _ignore_load_images):
+        docker_compose_yml = stringIO()
+        with mock.patch.dict(LocalSetup.SUPPORTED_VERSIONS, {'master': '8.0.0'}):
+            setup = LocalSetup(argv=self.common_setup_args + ["master", "--elasticsearch-enable-tls"])
+            setup.set_docker_compose_path(docker_compose_yml)
+            setup()
+        docker_compose_yml.seek(0)
+        got = yaml.load(docker_compose_yml)
+        services = set(got["services"])
+        self.assertIn("elasticsearch", services)
+
+        elasticsearch = got["services"]["elasticsearch"]
+        certs = "/usr/share/elasticsearch/config/certs/tls.crt"
+        certsKey = "/usr/share/elasticsearch/config/certs/tls.key"
+        caCerts = "/usr/share/elasticsearch/config/certs/ca.crt"
+        self.assertIn("xpack.security.http.ssl.enabled=true", elasticsearch["environment"])
+        self.assertIn("xpack.security.transport.ssl.enabled=true", elasticsearch["environment"])
+        self.assertIn("xpack.security.http.ssl.enabled=true", elasticsearch["environment"])
+        self.assertIn("xpack.security.http.ssl.enabled=true", elasticsearch["environment"])
+        self.assertIn("xpack.security.http.ssl.key=" + certsKey, elasticsearch["environment"])
+        self.assertIn("xpack.security.http.ssl.certificate=" + certs, elasticsearch["environment"])
+        self.assertIn("xpack.security.http.ssl.certificate_authorities=" + caCerts, elasticsearch["environment"])
+        self.assertIn("xpack.security.transport.ssl.key=" + certsKey, elasticsearch["environment"])
+        self.assertIn("xpack.security.transport.ssl.certificate=" + certs, elasticsearch["environment"])
+        self.assertIn("xpack.security.transport.ssl.certificate_authorities=" + caCerts, elasticsearch["environment"])
+        self.assertIn("curl -s -k https://localhost:9200/_cluster/health | grep -vq '\"status\":\"red\"'", elasticsearch["healthcheck"]["test"])
+
+    @mock.patch(cli.__name__ + ".load_images")
+    def test_kibana_tls(self, _ignore_load_images):
+        docker_compose_yml = stringIO()
+        with mock.patch.dict(LocalSetup.SUPPORTED_VERSIONS, {'master': '8.0.0'}):
+            setup = LocalSetup(argv=self.common_setup_args + ["master", "--kibana-enable-tls"])
+            setup.set_docker_compose_path(docker_compose_yml)
+            setup()
+        docker_compose_yml.seek(0)
+        got = yaml.load(docker_compose_yml)
+        services = set(got["services"])
+        self.assertIn("kibana", services)
+
+        kibana = got["services"]["kibana"]
+        certs = "/usr/share/kibana/config/certs/tls.crt"
+        certsKey = "/usr/share/kibana/config/certs/tls.key"
+        caCerts = "/usr/share/kibana/config/certs/ca.crt"
```

Comment:
Nit, not sure if possible, but I'd try to move this out to a constant, as it's repeated in many places.